DragaliaAPI/DragaliaAPI/Features/SavefileUpdate/V22Update.cs

```diff
@@ -0,0 +1,72 @@
+using DragaliaAPI.Database;
+using DragaliaAPI.Features.Player;
+using DragaliaAPI.Features.Present;
+using DragaliaAPI.Shared.Definitions.Enums;
+using DragaliaAPI.Shared.Features.Presents;
+using DragaliaAPI.Shared.MasterAsset;
+using DragaliaAPI.Shared.MasterAsset.Models.Story;
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore;
+
+namespace DragaliaAPI.Features.SavefileUpdate;
+
+/// <summary>
+/// Update to grant chapter 10 completion rewards to players that previously cleared chapter 10 and did not receive rewards.
+/// </summary>
+[UsedImplicitly]
+public class V22Update(
+    ApiContext apiContext,
+    IPresentService presentService,
+    IUserService userService,
+    ILogger<V22Update> logger
+) : ISavefileUpdate
+{
+    private const int Chapter10LastStoryId = 1001009;
+
+    private static readonly List<QuestStoryReward> Rewards = MasterAsset
+        .QuestStoryRewardInfo.Enumerable.Where(x => x.Id == Chapter10LastStoryId)
+        .First()
```

Comment:
Since the ID is the key I would use the O(1) lookup
```suggestion
        .QuestStoryRewardInfo[Chapter10LastStoryId]
```

Reply:
Ah this is much nicer!